### ISSUE TITLE
Hyperparameter study

### ISSUE THREAD

**Issue by amghr** [csml-beach/deep-bayesian-operator-nets#16]
The impact of hyperparameter choices (e.g., weights for interior loss and boundary condition loss) is only briefly mentioned. Detailed strategies (e.g., grid search, adaptive weighting) or providing the code would enhance reproducibility.

Do a GridSearchCv and providing code will suffice.

**Comment by amghr** [comment]
@elswit  I have started working on GridSearchCv and also on explaining how we ended up on those specific weights for optimal results during experiments manually.

**Comment by amghr** [comment]
@elswit Performed grid search for heat equation and uploaded the notebook: https://github.com/csml-beach/deep-bayesian-operator-nets/blob/main/notebooks/heat-equation/bayes-pinn-PDE-grid-search.ipynb

Below are the few things to note: 
1. Running grid search was expensive and time consuming. 
2. Grid search might not give us the exact weights as we can't just look at the total loss and there are so many other parameters to consider in PINNs before we call it an optimal solution.
3. The search space is very huge, especially for 2D equation and I am not sure on how to conduct grid search on that. 
4. Since the comment from reviewer says that either grid search or providing code would enhance reproducibility, we can write a reply saying that grid search would help us getting a rough estimate of weights and then we need to manually tune the hyper-parameters by looking at the solutions and graphs in order to obtain the best solution.

we can discuss the further steps in our next zoom call.

**Comment by elswit** [comment]
in the response letter you can be more elaborate about this extra run, say that u ran grid search, but then comment on why you might want an expert to look over the training : 
* We look at loss, but also spread of the ;learned parameter distribution

**Comment by amghr** [comment]
Added a new paragraph explaining the manual hyper-parameter turning, also added in response to reviews.
